### PR TITLE
feature: Add support for TF2.9.2 training images

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*Issue #, if available:*

*Description of changes:*

Add support for TF2.9.2 training images
*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by shreyapandit — APPROVED** [review]

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by vikas0203 — COMMENTED** [review]
/bot run all

**Review by vikas0203 — APPROVED** [review]

**Review by navinsoni — APPROVED** [review]
/bot run all

File: src/sagemaker/image_uri_config/tensorflow.json [hunk 1]
```diff
@@ -285,7 +285,8 @@
             "2.5": "2.5.1",
             "2.6": "2.6.3",
             "2.7": "2.7.0",
-            "2.8": "2.8.0"
+            "2.8": "2.8.0",
+            "2.9": "2.9.0"
```
Comment (mufaddal-rohawala) [inline src/sagemaker/image_uri_config/tensorflow.json:289]: @tejaschumbalkar training does not have a `2.9.0` image but a `2.9.1` image [here](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/image_uri_config/tensorflow.json#L1546). This causes the tests based on `tf_full_version` to fail. We have 2 options:

1. Can we switch to consistent TF Version for inference `2.9.1`?
2. Fix the failing tests, [current logic](https://github.com/aws/sagemaker-python-sdk/blob/ce7e4cdee655da386716ad48620b70d05a155064/tests/conftest.py#L440) is that it tries to get a minimum of latest training and inference versions and expects the versions to be consistent for training/inference.

**Review by mufaddal-rohawala — CHANGES_REQUESTED** [review]

File: src/sagemaker/image_uri_config/tensorflow.json (hunk 1 above)
Comment (tejaschumbalkar) [reply src/sagemaker/image_uri_config/tensorflow.json:289]: This PR was created when TF2.9.0 was released.

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run unit-tests

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run slow-tests, unit-tests

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run slow-tests

**Review by mufaddal-rohawala — APPROVED** [review]

**Pull request merged**